src/transxchange/read.rs

```diff
@@ -141,24 +142,54 @@ fn read_transxchange(transxchange: &Element, collections: &mut Collections) -> R
     unimplemented!()
 }
 
-fn read_transxchange_archive<P>(transxchange_path: P, collections: &mut Collections) -> Result<()>
+fn read_transxchange_from_zip<P>(transxchange_path: P, collections: &mut Collections) -> Result<()>
```

Comment:
you are in transxchange module so `read_from_path` is enough

same for `read_transxchange_from_zip`